components/molecule/modal/src/index.js

```diff
@@ -0,0 +1,216 @@
+import PropTypes from 'prop-types'
+import React, {Component} from 'react'
+import cx from 'classnames'
+import {SUPPORTED_KEYS} from './config'
+import {suitClass} from './helpers'
+
+class MoleculeModal extends Component {
+  constructor(...args) {
+    super(...args)
+
+    this.contentDOMEl = null
+    this.wrapperDOMEl = null
+    this.state = {
+      open: this.props.open
+    }
+  }
+
+  componentWillMount() {
+    document.addEventListener('keydown', this._onKeyDown)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this._onKeyDown)
+  }
+
+  _onKeyDown = event => {
+    if (this.state.open === false || this.props.closeOnEscKeyDown === false)
+      return
+    if (SUPPORTED_KEYS.includes(event.key)) {
```

Comment:
FYI: `includes` doesn't work in IE11 https://caniuse.com/#search=includes, but we have a polyfill for it https://github.com/SUI-Components/sui/blob/master/packages/sui-polyfills/src/index.js#L5, so no worries 👍